runtime/src/main/java/io/quarkiverse/cxf/CxfClientProducer.java

```diff
@@ -104,34 +138,106 @@ private void addToCols(String className, List<Interceptor<? extends Message>> co
         addToCols(className, CastUtils.<Interceptor> cast(cols), Interceptor.class);
     }
 
-    private <T> void addToCols(String className, List<T> cols, Class<T> clazz) {
+    private static <T> void addToCols(
+            String className,
+            List<T> cols,
+            Class<T> clazz) {
         Class<? extends T> cls;
         try {
             cls = Class.forName(className, false, Thread.currentThread().getContextClassLoader()).asSubclass(clazz);
         } catch (ClassNotFoundException | ClassCastException e) {
-            // silent failed
+            // silent fail
+            LOGGER.warn(format("no such class %s", className));
             return;
         }
         T item = null;
         try {
             item = CDI.current().select(cls).get();
-            if (item != null) {
-                cols.add(item);
-            }
         } catch (ClassCastException | UnsatisfiedResolutionException e) {
-            //silent fail
-        }
-        if (item != null) {
-            return;
+            // ignored
         }
         // if not found with beans just generate it.
 
         try {
-            item = cls.getConstructor().newInstance();
-            if (item != null) {
-                cols.add(item);
-            }
+            item = item == null ? cls.getConstructor().newInstance() : item;
         } catch (ReflectiveOperationException | RuntimeException e) {
+            /* ignore */
+        }
+
+        if (item == null) {
+            LOGGER.warn(format("unable to create instance of class %s", className));
+        } else {
+            cols.add(item);
+        }
+    }
+
+    /**
+     * Calculates the client info to use for producing a JAXWS client proxy.
+     *
+     * @param cxfConfig The current configuration
+     * @param ip Meta information about where injection of client proxy takes place
+     * @param meta The default to return
+     * @return not null
+     */
+    static private CXFClientInfo selectorCXFClientInfo(
```

Comment:
I like the idea to move selector here

Reply:
Question about selector do you handle default VS named client?

Reply:
> Question about selector do you handle default VS named client?

Don't understand the question? What is VS standing for?

Reply:
Versus

Reply:
Now I understand what VS means, not sure whether I got the question as such.

`selectorCXFClientInfo()` has three arguments, the current config, the injection point and the default meta information for a client of a given SEI. Basically, if no config applies then the meta information is returned. Otherwise a dedicated client info is returned based on given meta info. So in this sense `selectorCXFClientInfo()` handles a default and named client injection point.